students/alexLaws/lesson03/mailroom.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+donor_list = []
+donor_list.append({'Name': 'Ryan Moore', 'Donation': 500})
+donor_list.append({'Name': 'Ryan Moore', 'Donation': 250})
+donor_list.append({'Name': 'Ted Laws', 'Donation': 1000})
+donor_list.append({'Name': 'Ted Laws', 'Donation': 100})
+donor_list.append({'Name': 'Ben Snell', 'Donation': 150})
+donor_list.append({'Name': 'Andrew Crawford', 'Donation': 2000})
+donor_list.append({'Name': 'Andrew Crawford', 'Donation': 2000})
+donor_list.append({'Name': 'Andrew Crawford', 'Donation': 4000})
+donor_list.append({'Name': 'Beth Ross', 'Donation': 400})
+
+
+def user_action():
+    print("Choose an action from this list:")
+    print("Send a Thank You")
+    print("Create a Report")
+    print("Quit")
+    while True:
+        option = input("What is your choice: ")
+        if option.lower() in ["send a thank you", "create a report", "quit"]:
+            return option
+            break
+        else:
+            print("That is not an option. Choose again.")
+
+
+def dnrs(x):
+    names = []
+    for dnr in x:
+        person = dnr.get("Name")
+        if person in names:
+            pass
+        else:
+            names.append(person)
+    return names
+
+
+def send_to(x):
+    donors = dnrs(x)
+    recipient = input("Who is the thank you note for? Enter a full name or 'list' to see a list: ")
+    if recipient.lower() == 'list':
+        print("Here are all the donors:")
+        for name in donors:
+            print(name)
+        recipient = input("Who is the thank you note for? ")
+        return recipient
+    else:
+        return recipient
+
+
+def name_sum(person, x):
+    donor_total = 0
+    for i in x:
+        if i.get('Name') == person:
+            donor_total += int(i.get('Donation'))
+    return donor_total
+
+
+def name_cnt(person, x):
+    donor_count = 0
+    for i in x:
+        if i.get('Name') == person:
+            donor_count += 1
+    return donor_count
```

Comment:
The `name_sum` and `name_cnt` functions could be replaced by Python's built in `sum` and `len` functions if donation data were stored in a list.